### PR TITLE
Referral system changes

### PR DESCRIPTION
**Pull request by JasonSowers**

### CODE REVIEW

**Comment by JasonSowers** [comment]
do not merge yet as it will take a bit to test and get into production

File: BITCORNService/Utils/ReferralUtils.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using BITCORNService.Utils.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace BITCORNService.Utils
+{
+    public static class ReferralUtils
+    {
+        public static void UpdateReferralSync(BitcornContext dbContext, PlatformId platformId)
+        {
+            int? userId =0;
```
Comment (SamuliSalonen) [inline BITCORNService/Utils/ReferralUtils.cs]: this functionality already exists in util
`int? userId = BitcornUtils.GetUserForPlatform(platformId, dbContext).Select(u => u.UserId).FirstOrDefault();`

File: BITCORNService/Utils/ReferralUtils.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                            botWallet.Balance -= referrer.Amount;
+                            var stats = dbContext.UserStat.FirstOrDefault(s => s.UserId == referrer.UserId);
+                            if (stats != null)
+                            {
+                                stats.TotalReferralRewards += referrer.Amount;
+                            }
+                        }
+                    }
+                    userReferral.SyncDate = DateTime.Now;
+                }
```
Comment (SamuliSalonen) [inline BITCORNService/Utils/ReferralUtils.cs:49]: changes are never saved to the database

**Review by SamuliSalonen — CHANGES_REQUESTED** [review]

File: BITCORNService/Utils/ReferralUtils.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                case "twitter":
+                    userId = dbContext.UserIdentity.FirstOrDefault(u => platformId != null && u.TwitterId == platformId.Id)?.UserId;
+                    break;
+                case "discord":
+                    userId = dbContext.UserIdentity.FirstOrDefault(u => platformId != null && u.DiscordId == platformId.Id)?.UserId;
+                    break;
+                case "reddit":
+                    userId = dbContext.UserIdentity.FirstOrDefault(u => platformId != null && u.RedditId == platformId.Id)?.UserId;
+                    break;
+            }
```
Comment (SamuliSalonen) [inline BITCORNService/Utils/ReferralUtils.cs]: this functionality already exists in util
`int? userId = BitcornUtils.GetUserForPlatform(platformId, dbContext).Select(u => u.UserId).FirstOrDefault();`

**Review by SamuliSalonen — APPROVED** [review]
:shipit:

**Review by SamuliSalonen — APPROVED** [review]

**Pull request merged**